### PR TITLE
Updated Manifest and some SMI Changes

### PR DESCRIPTION
**Pull request by dhruv0000**
Co-authored-by: abishek <kumarabd@users.noreply.github.com>
Signed-off-by: dhruv0000 <patel.4@iitj.ac.in>

**Description**

This PR fixes: 
- The percentage calculations
- Fixed [#37](https://github.com/layer5io/meshery-adapter-library/issues/37)
- Updated smi-conformance service to `LoadBalancer`

**Notes for Reviewers**
- Need to change `dhruv0000/learn-layer5:smi` in `smi-conformance/manifest.yaml` to `layer5/learn-layer5:smi`, once approved.

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: smi-conformance/grpc/handlers.go [hunk 1]
```diff
@@ -56,15 +57,20 @@ func (s *Service) RunTest(ctx context.Context, req *conformance.Request) (*confo
 	}
 
 	result := test_gen.RunTest(config, req.Mesh.Annotations, req.Mesh.Labels)
-	totalcases := 3
-	failures := 0
+	totalSteps := 24
```
Comment (leecalcote) [inline smi-conformance/grpc/handlers.go:60]: Is this true for every mesh?

**Review by leecalcote — APPROVED** [review]

**Review by leecalcote — APPROVED** [review]
Other than `dhruv0000` image location, this LGTM

File: smi-conformance/grpc/handlers.go (hunk 1 above)
Comment (dhruv0000) [reply smi-conformance/grpc/handlers.go:60]: Aye. That are the total test Steps we are checking. Coincides with the no. of tests in [layer5.io](https://layer5.io/projects/service-mesh-interface-conformance).

File: smi-conformance/grpc/handlers.go [hunk 2]
```diff
@@ -89,17 +95,24 @@ func (s *Service) RunTest(ctx context.Context, req *conformance.Request) (*confo
 			}
 			d.Status = conformance.ResultStatus_FAILED
 			d.Capability = conformance.Capability_NONE
-			failures += 1
-			if (res.Assertions - failures) > (res.Assertions / 2) {
+
+			// A hacky way to see the testStep Failed, since KUDO only provides it in Failure.Message
+			re := regexp.MustCompile(`[0-9]+`)
+			stepFailed := re.FindAllString(res.Failure.Message, 1)
```
Comment (kumarabd) [inline smi-conformance/grpc/handlers.go:101]: check if `res.Failure!=nil`

File: smi-conformance/grpc/handlers.go [hunk 3]
```diff
@@ -89,17 +95,24 @@ func (s *Service) RunTest(ctx context.Context, req *conformance.Request) (*confo
...
+			passed, _ := strconv.Atoi(stepFailed[0])
+			passed = passed - 1
+			failures := stepsCount[res.Name] - passed
```
Comment (kumarabd) [inline smi-conformance/grpc/handlers.go:104]: checks for map not nil

File: smi-conformance/manifest.yml [hunk 4]
```diff
@@ -61,7 +61,7 @@ spec:
       serviceAccountName: meshery
       containers:
         - name: smi-conformance
-          image: "layer5/learn-layer5:smi"
+          image: "learn-layer5/learn-layer5:smi"
```
Comment (kumarabd) [inline smi-conformance/manifest.yml:64]: This should be `layer5/`, revert this @dhruv0000

File: smi-conformance/grpc/handlers.go (hunk 2 above)
Comment (dhruv0000) [reply smi-conformance/grpc/handlers.go:101]: Aready checked [here](https://github.com/layer5io/learn-layer5/pull/76/files#diff-baad359c237d9e0cd128eb91bc99f1336175b3ea7d45d7f2658ae49a0d752d2bR83) in `if len(res.Failure.Text) > 2`

Comment (kumarabd) [reply smi-conformance/grpc/handlers.go:101]: No, res.Failure nil will result in panic.

Comment (dhruv0000) [reply smi-conformance/grpc/handlers.go]: Okey. Added the panic test.

**Review by kumarabd — APPROVED** [review]

**Pull request merged**